addons/pos_sale/tests/test_pos_sale_flow.py

```diff
@@ -695,6 +696,14 @@ def test_downpayment_with_taxed_product(self):
         self.assertEqual(invoice_pdf_content.count('Product B'), 1)
         self.assertEqual(invoice_pdf_content.count('Product C'), 1)
 
+        # the last 3 lines should be down payment lines, and should have
+        # the date formatted according to user's language (in this test, it's en_US)
+        for order_line in sale_order.order_line[-3:]:
```

Comment:
Might be better to get the downpayment lines based on the product. You could do something like `sale_order.order_line.filtered(lambda ol: ol.product_id.id == self.downpayment_product.id)`